File: content/problems/1287-element-appearing-more-than-25-in-sorted-array/solutions/sorted-run-counting/Solution.java
class Solution {
    public int findSpecialInteger(int[] arr){
        for(int start=0;start<arr.length;){
            int end=start+1;
            while(end<arr.length&&arr[end]==arr[start])end++;
            if((end-start)*4>arr.length)return arr[start];
            start=end;
        }
        return -1;
    }
}

File: content/problems/1287-element-appearing-more-than-25-in-sorted-array/solutions/sorted-run-counting/Solution.kt
class Solution {
    fun findSpecialInteger(arr:IntArray):Int{
        var start=0;
        while(start<arr.size){
            var end=start+1;
            while(end<arr.size&&arr[end]==arr[start])end++;
            if((end-start)*4>arr.size)return arr[start];
            start=end
        };
        return -1
    }
}
